sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -207,6 +207,8 @@ private[hive] class HiveClientImpl(
   /** Returns the configuration for the current session. */
   def conf: HiveConf = state.getConf
 
+  private val userName = state.getAuthenticator.getUserName
```

Comment:
All the `SparkSession`s share the same internal Hive `SessionState`. After we create it, it will be set, if my understanding is not wrong, no matter which `HiveAuthenticationProvider` that users choose. 

Previously, before merging the PR https://github.com/apache/spark/pull/17311, do we have the same issue?

cc @vanzin @dongjoon-hyun @yhuai